nixnet/types.py

```diff
@@ -495,6 +496,130 @@ def __repr__(self):
             self.rx_err_count)
 
 
+class LinFrame(object):
+    """LIN Frame.
+
+    Attributes:
+        identifier(int): LIN frame arbitration identifier.
+        echo(bool): If the frame is an echo of a successful
+            transmit rather than being received from the network.
+        type(:any:`nixnet._enums.FrameType`): Frame type.
+        timestamp(int): Absolute time the XNET interface received the end-of-frame.
+        eventslot(bool): Whether the frame was received within an
+            event-triggered slot or an unconditional or sporadic slot.
+        eventid(int): Identifier for an event-triggered slot.
+        payload(bytes): A byte string representing the payload.
+    """
+
+    __slots__ = [
+        "identifier",
+        "echo",
+        "type",
+        "timestamp",
+        "eventslot",
+        "eventid",
+        "payload"]
+
+    _FRAME_ID_MASK = 0x0000003F
+
+    def __init__(self, identifier, type=constants.FrameType.LIN_DATA, payload=b""):
+        # type: (int, constants.FrameType, bytes) -> None
+        self.identifier = identifier
+        self.echo = False  # Used only for Read
+        self.type = type
+        self.timestamp = 0  # Used only for Read
+        self.eventslot = False  # Used only for Read
+        self.eventid = 0  # Used only for Read
+        self.payload = payload
+
+    @classmethod
+    def from_raw(cls, frame):
+        # type: (RawFrame) -> LinFrame
+        """Convert from RawFrame.
+
+        >>> raw = RawFrame(5, 2, constants.FrameType.LIN_DATA, _cconsts.NX_FRAME_FLAGS_TRANSMIT_ECHO, 0, b'\x01')
+        >>> LinFrame.from_raw(raw)
+        LinFrame(identifier=2, echo=True, timestamp=0x5, len(payload)=1)
+        """  # NOQA: E501
+        identifier = frame.identifier & cls._FRAME_ID_MASK
+        lin_frame = LinFrame(identifier, constants.FrameType(frame.type), frame.payload)
+        lin_frame.timestamp = frame.timestamp
+        lin_frame.echo = bool(frame.flags & _cconsts.NX_FRAME_FLAGS_TRANSMIT_ECHO)
+        lin_frame.eventslot = bool(frame.flags & _cconsts.NX_FRAME_FLAGS_LIN_EVENT_SLOT)
+        if lin_frame.eventslot:
+            lin_frame.eventid = frame.info
+        else:
+            lin_frame.eventid = 0
+
+        return lin_frame
+
+    def to_raw(self):
```

Comment:
A couple parts of `to_raw` are missing coverage
https://coveralls.io/builds/12752500/source?filename=nixnet%2Ftypes.py#L556